### PR TITLE
Cleanup dependencies

### PR DESCRIPTION
**Pull request by stanlemon**

### CODE REVIEW

File: src/main/java/com/salesforce/storm/spout/dynamic/Tools.java [hunk 1]
```diff
@@ -49,14 +47,8 @@
      * @return sshallow cloned map that is immutable.
      */
     public static <K,V> Map<K,V> immutableCopy(Map<K,V> sourceMap) {
-        // If we're already dealing with an UnmodifiableMap
-        if (sourceMap instanceof UnmodifiableMap) {
```
Comment (stanlemon) [inline src/main/java/com/salesforce/storm/spout/dynamic/Tools.java]: Interestingly this is NOT the `UnmodifiableMap` used by `Collections.unmodifiableMap()` so this never actually did anything.

File: src/main/java/com/salesforce/storm/spout/documentation/DocGenerator.java [hunk 2]
```diff
@@ -117,7 +117,7 @@ private void injectContent(final String content) throws IOException {
         Files.copy(inputPath, tempOutPath);
 
         try (BufferedReader readmeReader = Files.newBufferedReader(inputPath, StandardCharsets.UTF_8);
-            PrintWriter readmePrintWriter = new PrintWriter(Files.newBufferedWriter(tempOutPath, StandardCharsets.UTF_8))) {
+             PrintWriter readmePrintWriter = new PrintWriter(Files.newBufferedWriter(tempOutPath, StandardCharsets.UTF_8))) {
```
Comment (Crim) [inline src/main/java/com/salesforce/storm/spout/documentation/DocGenerator.java]: extra space here?

File: src/main/java/com/salesforce/storm/spout/dynamic/Tools.java (hunk 1 above)
Comment (Crim) [reply src/main/java/com/salesforce/storm/spout/dynamic/Tools.java]: should we add the correct instance check here?

File: src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java [hunk 3]
```diff
@@ -43,12 +42,11 @@
 public class LogRecorder implements MetricsRecorder {
 
     private static final Logger logger = LoggerFactory.getLogger(LogRecorder.class);
-    private final Map<String, Long> counters = Maps.newConcurrentMap();
-    private final Map<String, CircularFifoQueue> averages = Maps.newConcurrentMap();
```
Comment (Crim) [inline src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java]: Was averages just not used?

File: src/main/java/com/salesforce/storm/spout/dynamic/Tools.java (hunk 1 above)
Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/Tools.java]: I looked into this, the problem is the actual class is private to the Collections class, so we can’t.

Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/Tools.java]: For reference http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/Collections.java#l1499

File: src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java (hunk 3 above)
Comment (stanlemon) [reply src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java]: Correct, which made this one easy to deal with. 😎

**Pull request merged**